rllib/env/single_agent_env_runner.py

```diff
@@ -249,6 +249,8 @@ def _sample_timesteps(
                     observation=obs[env_index],
                     infos=infos[env_index],
                 )
+            self._was_terminated = [False for _ in range(self.num_envs)]
```

Comment:
This is completely new auto-reset logic of gymnasium 1.0. The sub-env only gets reset'd upon the next(!) step call (with a fake reward of 0.0 and term/trunc=guaranteed False; and the obs/infos being the reset-obs/infos).
This is actually good for us as we should always do the env-to-module connector pass (even after the last timestep with the terminal obs in the Episodes list) to make sure the user - in case they are writing to the episode - gets a chance to also alter the final obs.